components/EditPublicMessagePopup.js

```diff
@@ -0,0 +1,299 @@
+      setEditMessagePopupPosition({ x, y: cardY, right: true });
+      // /** Check if popup fits to the left */
+    } else if (cardRect.x - popupRect.width - POPUP_ARROW_WIDTH >= 0) {
+      const x = cardX - popupRect.width - POPUP_ARROW_WIDTH;
+      setEditMessagePopupPosition({ x, y: cardY, left: true });
+      // /** Position popup at the bottom */
+    } else {
+      const popupHalfWidth = popupRect.width / 2;
+      const cardHalfWidth = cardRect.width / 2;
+      let x = cardX - popupHalfWidth + cardHalfWidth;
+      /** Add vertical offset */
+      const y = cardY + cardRect.height + 10;
+      /** Check if popup will fit horizontally. Adjust if it doesn't */
+      if (x < 0) {
+        x = 0;
+      } else if (x + popupRect.width > viewportWithWithoutScrollbar) {
+        x -= x + popupRect.width - viewportWithWithoutScrollbar;
+      }
+      setEditMessagePopupPosition({ x, y });
+    }
+  }, []);
+
+  /** Lazily setup the root element */
+  function getRootElem() {
+    if (!rootElementRef.current) {
+      rootElementRef.current = document.createElement('div');
+    }
+    return rootElementRef.current;
+  }
+
+  return createPortal(
+    <Mutation mutation={EditPublicMessageMutation}>
+      {(submitMessage, { loading, error }) => (
+        <EditPublicMessagePopupContainer
+          data-cy="EditPublicMessagePopup"
+          position={editMessagePopupPosition}
+          ref={popupRef}
+        >
+          <Flex justifyContent="flex-end">
+            <Times size="1em" color="#a2a2a2" cursor="pointer" onClick={onClose} />
+          </Flex>
+          <Flex flexDirection="column" p={2}>
+            <Span fontSize="Paragraph" color="black.600" mb={2}>
+              <FormattedMessage id="contribute.publicMessage" defaultMessage="Leave a public message (Optional)" />
+            </Span>
+
+            <StyledInput
+              name="publicMessage"
+              as="textarea"
+              px={10}
+              py={10}
+              width={240}
+              height={112}
+              fontSize="Paragraph"
+              style={{ resize: 'none' }}
+              placeholder={intl.formatMessage(messages.publicMessagePlaceholder)}
+              value={messageDraft}
+              maxLength={PUBLIC_MESSAGE_MAX_LENGTH}
+              onChange={e => setMessageDraft(e.target.value)}
+              disabled={loading}
+            />
+            {error && (
+              <Span color="red.500" fontSize="Caption" mt={2}>
+                {error}
+              </Span>
+            )}
+            <Box m="0 auto">
+              <StyledButton
+                data-cy="EditPublicMessagePopup_SubmitButton"
+                buttonSize="small"
+                fontWeight="bold"
+                px={4}
+                mt={3}
+                onClick={async () => {
+                  await submitMessage({
+                    variables: {
+                      FromCollectiveId: +fromCollectiveId,
```

Comment:
When I was testing locally sometime I got `fromCollectiveId` as string and that was causing an error on GraphQL:

![Screen Shot 2019-10-27 at 10 31 00 PM](https://user-images.githubusercontent.com/1315054/67651014-915e0c00-f915-11e9-8d93-d722a0aa7e90.png)

I think both `fromCollectiveId` and `collectiveId` should always be integer, since that what they are in the database but that was not directly related to this issue so I didn't spent the time to find the reason why it was a string.